src/Discord.Net.Commands/Attributes/Preconditions/RequireUserPermissionAttribute.cs

```diff
@@ -62,7 +62,7 @@ public override Task<PreconditionResult> CheckPermissionsAsync(ICommandContext c
                 if (guildChannel != null)
                     perms = guildUser.GetPermissions(guildChannel);
                 else
-                    perms = ChannelPermissions.All(guildChannel);
+                    perms = ChannelPermissions.All(context.Channel);
```

Comment:
Wouldn't it be better to refactor this into `if (context.Channel is IGuildChannel guildChannel)`?